client/asset/interface.go

```diff
@@ -10,26 +10,46 @@ import (
 	"decred.org/dcrdex/dex"
 )
 
+// WalletConfig is the configuration settings for the wallet.
+type WalletConfig struct {
+	// Account is the name of the dcrwallet account. The account and wallet must
+	// already exist.
+	Account string
+	// INIPath is the path of a dcrwallet-like configuration file. RPC credentials
+	// will be read from the file. You can provide the path to your actual
+	// dcrwallet configuration file.
+	INIPath string
+	// TipChange is a function that will be called when the blockchain monitoring
+	// loop detects a new block. If the error supplied is nil, the client should
+	// check the confirmations on any negotiating swaps to see if action is
+	// needed. If the error is non-nil, the wallet monitoring loop encountered an
+	// error while retreiving tip information.
+	TipChange func(error)
+}
+
 // Wallet is a common interface to be implemented by cryptocurrency wallet
 // software.
 type Wallet interface {
+	dex.Runner
+	// Connect connects the wallet to the RPC server.
+	Connect() error
 	// Balance should return the total available funds in the wallet.
 	// Note that after calling Fund, the amount returned by Balance may change
 	// by more than the value funded.
-	Balance() (available, locked uint64, err error)
+	Balance(*dex.Asset) (available, locked uint64, err error)
```

Comment:
Adding the `*dex.Asset` argument to the methods directly enables a single wallet to be used with any number of DEX accounts.